src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepDynamicContext.java

```diff
@@ -0,0 +1,246 @@
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.support.DefaultStepContext;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Persistent representation for context of {@link ExecutorStepExecution}.
+ * Supersedes {@link FilePathDynamicContext} (never mind {@link org.jenkinsci.plugins.workflow.support.pickles.FilePathPickle}),
+ * {@link org.jenkinsci.plugins.workflow.support.pickles.ExecutorPickle},
+ * {@link org.jenkinsci.plugins.workflow.support.pickles.ComputerPickle},
+ * and {@link org.jenkinsci.plugins.workflow.support.pickles.WorkspaceListLeasePickle}.
+ */
+@Restricted(NoExternalUse.class)
+public final class ExecutorStepDynamicContext implements Serializable {
+
+    private static final Logger LOGGER = Logger.getLogger(ExecutorStepDynamicContext.class.getName());
+
+    private static final long serialVersionUID = 1;
+
+    final @NonNull ExecutorStepExecution.PlaceholderTask task;
+    final @NonNull String node;
+    private final @NonNull String path;
+    /** Non-null after {@link #resume} if all goes well. */
+    private transient @Nullable Executor executor;
+    /** Non-null after {@link #resume} if all goes well. */
+    private transient @Nullable WorkspaceList.Lease lease;
+
+    ExecutorStepDynamicContext(ExecutorStepExecution.PlaceholderTask task, WorkspaceList.Lease lease, Executor executor) {
+        this.task = task;
+        this.node = FilePathUtils.getNodeName(lease.path);
+        this.path = lease.path.getRemote();
+        this.executor = executor;
+        this.lease = lease;
+    }
+
+    void resume(StepContext context) throws Exception {
+        if (executor != null) {
+            throw new IllegalStateException("Already resumed");
+        }
+        while (Jenkins.get().getInitLevel() != InitMilestone.COMPLETED) {
+            LOGGER.fine(() -> "waiting to schedule task for " + path + " on " + node + " until Jenkins completes startup");
+            Thread.sleep(100);
+        }
+        Queue.Item item = Queue.getInstance().schedule2(task, 0).getItem();
+        if (item == null) {
+            // TODO should also report when !ScheduleResult.created, since that is arguably an error
+            throw new IllegalStateException("queue refused " + task);
+        }
+        LOGGER.fine(() -> "scheduled " + item + " for " + path + " on " + node);
+        TaskListener listener = context.get(TaskListener.class);
+        if (!node.isEmpty()) { // unlikely to be any delay for built-in node anyway
+            listener.getLogger().println("Waiting for reconnection of " + node + " before proceeding with build");
+        }
+        Queue.Executable exec;
+        try {
+            exec = item.getFuture().getStartCondition().get(ExecutorStepExecution.TIMEOUT_WAITING_FOR_NODE_MILLIS, TimeUnit.MILLISECONDS);
```

Comment:
What if Jenkins is started in quiet down mode?

Does this mean that the job that was running will always fail?